### DISCUSSION TITLE
Scheduled job not running from windows service but runs from hosted app in visual studio.

### DISCUSSION THREAD

**Discussion by jockorob** [quartznet/quartznet#1898] [Q&A]
I'm new to quartz. I have a .net framework 4.8 windows service. When I host the service code via a windows forms app and debug it in visual studio my scheduled job runs. However when I install the windows service the job doesn't run. I have logging that shows that the job schedules successfully but doesn't run. Any ideas what I need to change to get it to work from a windows service? I'm assuming there is something wrong with the IScheduler instance. 

ISchedulerFactory schedulerFactory = new StdSchedulerFactory();
_quartzScheduler = schedulerFactory.GetScheduler().GetAwaiter().GetResult();
_quartzScheduler.Start().GetAwaiter().GetResult();
ScheduleSimpleJob();

**Comment by lahma** [comment]
Have you checked [the server example](https://github.com/quartznet/quartznet/tree/main/src/Quartz.Server)? It contains a sample that can be installed as Windows service and has logging.

**Reply by jockorob** [reply]
Thanks for the help! I'm not using topshelf so my code looks quite a bit different. The program.cs of my windows service project looks like this.
static class Program
    {
        static void Main()
        {
            ServiceBase[] ServicesToRun;
            ServicesToRun = new ServiceBase[] 
			{ 
				new ClientWindowsService() 
			};
            ServiceBase.Run(ServicesToRun);
        }
    }

The ClientWindowsService class inherits from ServiceBase. When it starts I create the StdSchedulerFactory, call GetScheduler(), start it, and then schedule my job as shown in my last post. When I run that code from a console app the job runs. When I run it from my windows service it does not.

**Reply by lahma** [reply]
Maybe you should use such a ready-made library that handles all the specifics and also allows to run as console cleanly?

**Reply by jockorob** [reply]
Unfortunately that is not an option. This is not a greenfield project. The code has been working this way for over 10 years. I'm tasked with  adding some scheduler functionality to an existing windows service and thought quartz would be a good fit.

**Reply by jockorob** [reply]
I downgraded to version 2.6.2, removed the async code as needed (e.g. GetAwaiter().GetResult()) and my job runs as expected from the windows service. Is there any danger in using 2.6.2? Does this shed any light onto what could be going wrong in 3.5.0?

**Reply by lahma** [reply]
Some have had trouble with the diagnosticssource dependency, you can search the issues. 2.x isn't supported anymore but if it works for you,  why not.
